pkg/satellite/stream/stream.go

```diff
@@ -296,6 +320,9 @@ func (ss *satelliteStream) recvLoop() {
 				}
 				payload := telemetry.Data
 				log.Debug("received data: streamId: %v, planId: %s, framing type: %s, size: %d bytes\n", ss.streamId, planId, telemetry.Framing, len(payload))
+				if ss.enableAutoClose && len(payload) == 0 {
```

Comment:
I'd appreciate a comment here (or maybe there's a better place) that explains what's going on here. It might not be evident that a message with an empty payload denotes the end of a stream.

Also, it may be easier to separate this case outside of the `for` loop. Something like `if ss.enableAutoClose && len(telemetry) == 1 && len(telemetry.Data) == 0` so it is a very explicit check.

Reply:
Great idea to be more explicit here.

Reply:
And, for users, a bit more background in the docs about what this 'end of stream' message is and when it occurs.

Reply:
@chelliwell We mistakenly put mention of the 'end of stream' message in our help messaging, which we think is confusing for the users. The implementation detail is not important for users to understand when using the CLI.

Reply:
> @chelliwell We mistakenly put mention of the 'end of stream' message in our help messaging, which we think is confusing for the users. The implementation detail is not important for users to understand when using the CLI.

Ok. One question though please: the release notes say "which is normally the plan end time plus some buffer". Are there any 'abnormal' scenarios where it _won't_ close then?

Reply:
There are cases where the END message may not be delivered if certain internal systems have a failure. We recommend to build in redundancy into your integration in the event that the CLI does not auto close.